File: client/src/app/core/dto/task/ph-task.dto.ts
import { Guid } from 'guid-typescript';

export class PhTaskDto {
  Id: Guid = Guid.createEmpty();
  SerialNumber: number = 0;
  Name: string = '';
  Notes?: string;
  CreatedAt: Date | string | number = new Date();
  CreatedById: Guid = Guid.createEmpty();
  UpdatedAt?: Date | string | number;
  UpdatedById?: Guid;
  StatusId: Guid = Guid.createEmpty();
}

File: api/Core/Dto/Task/PhTaskDto.cs
namespace ProductivityHarborApi.Core.Dto.Task
{
    public class PhTaskDto
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public int SerialNumber { get; set; }
        public Guid StatusId { get; set; }
        public string? Notes { get; set; }
        public virtual DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.Now;
        public virtual Guid? CreatedByUserId { get; set; }
        public virtual DateTimeOffset? UpdatedAt { get; set; }
        public virtual Guid? UpdatedByUserId { get; set; }
    }
}
